enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java

```diff
@@ -0,0 +1,124 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.metrics;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.UUID;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.graphdb.factory.GraphDatabaseFactory;
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.helpers.Settings;
+import org.neo4j.helpers.collection.MapUtil;
+import org.neo4j.test.TargetDirectory;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.junit.Assert.assertThat;
+import static org.neo4j.metrics.MetricsSettings.CsvFile.single;
+import static org.neo4j.metrics.MetricsSettings.csvEnabled;
+import static org.neo4j.metrics.MetricsSettings.csvFile;
+import static org.neo4j.metrics.MetricsSettings.csvPath;
+
+public class MetricsExposedListDocIT
+{
+
+    @Rule
+    public final TargetDirectory.TestDirectory folder = TargetDirectory.testDirForTest( getClass() );
+
+    GraphDatabaseService db;
+    private File outputFile;
+
+    @Before
+    public void setup() throws IOException
+    {
+        String dbPath = folder.directory( "data" ).getAbsolutePath();
+        outputFile = folder.file( "metrics.csv" );
+        db = new GraphDatabaseFactory().newEmbeddedDatabaseBuilder( dbPath ).
+                setConfig( GraphDatabaseSettings.allow_store_upgrade, Settings.TRUE ).
+                setConfig( csvEnabled, Settings.TRUE ).
+                setConfig( csvFile, single.name() ).
+                setConfig( csvPath, outputFile.getAbsolutePath() ).newGraphDatabase();
+    }
+
+    @After
+    public void shutdown()
+    {
+        db.shutdown();
+    }
+
+
+    @Test
+    public void listExposedMetrics() throws Exception
+    {
+        // Create some activity that will generate some metrics data.
+        for ( int i = 0; i < 1000; i++ )
+        {
+            try ( Transaction tx = db.beginTx() )
+            {
+                Map<String,Object> params = MapUtil.map( "name", UUID.randomUUID().toString() );
+                db.execute( "create (n:Label {name: {name}})", params );
+                tx.success();
+            }
+        }
+
+        // Read the metrics we recorded
+        String[] headers;
+        try ( BufferedReader reader = new BufferedReader( new FileReader( outputFile ) ) )
+        {
+            // First verify that some metrics have been written
+            headers = reader.readLine().split( "," );
+            assertThat( headers[0], is( "timestamp" ) );
```

Comment:
I am afraid this is not gonna be always true. Can't we search for that header instead?